File: javapoet/common/src/main/java/com/flannery/common/base/BaseActivity.java
package com.xiangxue.common.base;

import android.os.Bundle;
import android.util.Log;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.xiangxue.common.utils.Cons;

/**
 * 项目父Activity
 */
public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e(Cons.TAG, "common/BaseActivity");
    }
}

File: gradledemo/gradledownload/src/main/java/com/fannery/gradledownload/BaseActivity.kt
package com.fannery.gradledownload

import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.lang.StringBuilder

/**
 * Time:2021/7/26 10:48
 * Author:
 * Description:
 */
open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)

        printStackTrace()
    }


    fun printStackTrace() {
        // StackTraceElement[] trace = getOurStackTrace();
        // for (StackTraceElement traceElement : trace)
        // s.println("\tat " + traceElement);
        val sb = StringBuilder()
        for (element in Thread.currentThread().stackTrace) {
            sb.append("\tat").append(element).appendLine()
        }
        Toast.makeText(this, sb.toString(), Toast.LENGTH_LONG).show()
        Log.e("MainActivity", sb.toString())
    }


}